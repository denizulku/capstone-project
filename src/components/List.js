import React, { useState } from "react";
import styled from "styled-components";

export default function List({ habits }) {
  return (
    <section>
      {habits.map((habit) => (
        <div>{habit}</div>
      ))}
    </section>
  );
}
