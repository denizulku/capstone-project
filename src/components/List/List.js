import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import Habit from '../Habit/HabitItem'

export default function List({ habits, item }) {
  return (
    <HabitList>
      {habits.map((habit) => (
        <div key={item}>
          <HabitContainer>
            <Habit text={habit} />
          </HabitContainer>
        </div>
      ))}
    </HabitList>
  )
}

const HabitList = styled.section`
  margin-top: 20px;
  position: relative;
  text-align: center;
`
const HabitContainer = styled.div`
  display: inline-block;
  margin-top: 10px;
  width: 260px;
  height: 25px;
  background-color: #ffc179;
  border-radius: 10px;
`
