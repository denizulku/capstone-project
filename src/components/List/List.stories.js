import React from 'react'
import List from './List'

export default {
  title: 'List',
  component: List,
}

const Template = (args) => <List {...args} />

export const Primary = Template.bind({})

Primary.args = {
  habits: ['foo', 'bar'],
}

Primary.argTypes = { onClick: { action: 'checkHabit' } }
