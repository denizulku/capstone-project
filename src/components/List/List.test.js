import React from 'react'
import { render } from '@testing-library/react'
import List from './List'

describe(List, () => {
  const today = new Date()
  const months = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
  ]
  it('shows the current month', () => {
    expect(`${months[today.getMonth()]}`)
  })
})
