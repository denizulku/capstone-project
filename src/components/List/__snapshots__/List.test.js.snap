// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `[Function]`;

exports[`renders correctly 1`] = `[Function]`;

exports[`shows correct date 1`] = `[Function]`;
